Sources/HaCWebsiteLib/NewWorkshopManager.swift

```diff
@@ -17,15 +17,40 @@ public struct NewWorkshopManager {
     attributes: .concurrent
   )
 
+  private static var pollSource: DispatchSourceTimer! = nil
+  private static var currentlyUpdating = false
+
   /// The list of Workshops as derived from the workshop repos last time they were processed. Keys are workshop repo names
   public private(set) static var workshops: [String: NewWorkshop] = [:]
 
   /// All the workshop repos that we know about. Keys are workshop repo names
   private static var workshopRepoUtils: [String: GitUtil] = [:]
 
   public static func update() {
-    updateRepoList()
-    updateWorkshopsData()
+    print("Updating workshops...")
+    // Prevent multiple updates at once (might happen if an update takes longer than the poll interval)
+    if !currentlyUpdating {
+      currentlyUpdating = true
+      updateRepoList()
+      updateWorkshopsData()
+      currentlyUpdating = false
+    }
+    print("Workshops updated")
+  }
+
+  public static func startPoll() {
+    let pollQueue = DispatchQueue(label: "com.hac.website.workshops-poll-timer", attributes: .concurrent)
```

Comment:
Not a blocker but might be nice to abstract this polling behaviour away. If only `RxSwift` worked 😢